lib/LMSManagers/LMSHelpdeskManager.php

```diff
@@ -1503,4 +1514,26 @@ public function CheckTicketAccess($ticketid) {
 			LIMIT 1',
 			array($userid, $ticketid, $userid, $ticketid));
 	}
+
+    public function GetRelatedTicketIds($ticketid) {
+        return $this->db->GetAll('SELECT id FROM rttickets WHERE parentid = ?', array($ticketid));
+    }
+    
+    public function GetTicketParentID($ticketid) {
+	    if(empty($ticketid))
+	        return $this->db->GetOne('SELECT parentid FROM rttickets WHERE id = ?', array($ticketid));
+	    else
+	        return null;
+    }
+
+    public function IsTicketLoop($ticketid, $parentid) {
+        global $LMS;
+
+        if ($ticketid == $parentid)
+            return 1;
+        if (empty($parentid))
+            return null;
+        $parentid = $LMS->GetTicketParentID($parentid);
+        $LMS->IsTicketLoop($ticketid,$parentid);
```

Comment:
Pytam, bo nie jestem tego pewien: nie lepiej tu zrobić:
```php
$this->isTicketLoop($ticketid, $parentid);
```
niż robić przez globalny obiekt $LMS? Mało elegancko i mniej wydajnie.

Reply:
Jeśli zadziała to proszę bardzo.
Tutaj chodzi o to żeby wywołać funkcję z niej samej.